File: crates/core/script/templates/snippets/callbacks.lua
on_init = function(ctx)
  ctx.log("info", "script initialized (" .. tostring(GC_TEMPLATE) .. ")")
end,

on_update = function(ctx, delta)
  local _ = ctx
  local _ = delta
  -- Remove this callback when your script is event-driven only.
end,

on_event = function(ctx, event)
  if event.kind == "paramChanged" then
    local payload = event.payload or {}
    local _ = payload
    ctx.log("info", "paramChanged from node " .. tostring(event.origin))
  elseif event.kind == "custom" then
    ctx.log("info", "custom event received")
  end
end,

on_destroy = function(ctx)
  ctx.log("info", "script destroyed")
end,

File: crates/core/script/templates/snippets/callbacks.js
function init() {
  log(`script initialized (${String(GC_TEMPLATE)})`);
}

function update(delta) {
  void delta;
  // Remove this callback when your script is event-driven only.
}

function event(event) {
  if (event.kind === "paramChanged") {
    const payload = event.payload ?? {};
    void payload;
    log(`paramChanged from node ${String(event.origin)}`);
  } else if (event.kind === "custom") {
    log("custom event received");
  }
}

function destroy() {
  log("script destroyed");
}
